internal/service/dns_config/api_delegation_resource_test.go

```diff
@@ -0,0 +1,446 @@
+			// Create and Read
+			{
+				Config: testAccDelegationFqdn("test.123.", "123."),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDelegationExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "fqdn", "test.123."),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccDelegationFqdn("test1.123.", "123."),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDelegationExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "fqdn", "test1.123."),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccDelegationResource_Tags(t *testing.T) {
+	var resourceName = "bloxone_dns_delegation.test_tags"
+	var v dns_config.ConfigDelegation
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccDelegationTags("test.123.", "123.", map[string]string{
+					"tag1": "value1",
+					"tag2": "value2",
+				}),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDelegationExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "tags.tag1", "value1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.tag2", "value2"),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccDelegationTags("test.123.", "123.", map[string]string{
+					"tag2": "value2changed",
+					"tag3": "value3",
+				}),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDelegationExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "tags.tag2", "value2changed"),
+					resource.TestCheckResourceAttr(resourceName, "tags.tag3", "value3"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccDelegationResource_View(t *testing.T) {
+	var resourceName = "bloxone_dns_delegation.test_view"
+	var v dns_config.ConfigDelegation
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccDelegationView("test.123.", "123.", "bloxone_dns_view.test"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDelegationExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttrPair(resourceName, "view", "bloxone_dns_view.test", "id"),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccDelegationView("test.123.", "123.", "bloxone_dns_view.test1"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDelegationExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttrPair(resourceName, "view", "bloxone_dns_view.test1", "id"),
```

Comment:
can you check here if the previous object is destroyed here, since we use require_replace